src/site/_includes/partials/analytics.njk

```diff
@@ -1,14 +1,12 @@
 {% set analyticsScript %}
-window.ga =
-  window.ga ||
-  function () {
-    (ga.q = ga.q || []).push(arguments);
-  };
-ga.l = +new Date();
-ga('create', '{{ site.analytics.ids.prod }}');
-ga('set', 'transport', 'beacon');
-ga('set', 'anonymizeIp', true);
-ga('set', 'page', window.location.pathname);
-ga('set', '{{ site.analytics.dimensions.TRACKING_VERSION }}', '{{ site.analytics.version}}');
+window.dataLayer = window.dataLayer || [];
+function gtag(){dataLayer.push(arguments);}
+gtag('js', new Date());
+{% if (site.env != 'prod') %}
+  loadScript('https://www.googletagmanager.com/gtag/js?id={{ site.analytics.ids.ga4 }}', null);
```

Comment:
@devnook FYI, in your cookie baner changes PR, loading this script should be gated on users having given consent. That seems like the easiest way to ensure no data is sent for users who haven't consented.

Reply:
Also (sorry if this is obvious) but FYI loading the script doesn't need to happen here, it could happen at any point after the three lines above have run.

```js
window.dataLayer = window.dataLayer || [];
window.ga ||	function gtag(){dataLayer.push(arguments);}
function () {	gtag('js', new Date());
```